framework/src/positions/ParsedSelectionPositions.C

```diff
@@ -0,0 +1,168 @@
+{
+  InputParameters params = Positions::validParams();
+  params += NonADFunctorInterface::validParams();
+  params += BlockRestrictable::validParams();
+  params += FunctionParserUtils<false>::validParams();
+
+  // Parsed expression parameters
+  params.addRequiredCustomTypeParam<std::string>(
+      "expression",
+      "FunctionExpression",
+      "Parsed function expression to compute the selection criterion. Note that x,y,z and t can be "
+      "used in the expression without being declared as functors.");
+  params.addParam<std::vector<MooseFunctorName>>(
+      "functor_names", {}, "Functors to use in the parsed expression");
+  params.addParam<std::vector<std::string>>(
+      "functor_symbols",
+      {},
+      "Symbolic name to use for each functor in 'functor_names' in the parsed expression. If not "
+      "provided, then the actual functor names will be used in the parsed expression.");
+
+  // Use the default behaviors of a Positions object
+  // Sort positions as usual
+  params.set<bool>("auto_sort") = true;
+  // Need to synchronize positions across ranks
+  params.set<bool>("auto_broadcast") = true;
+
+  // Keep as up-to-date as possible given the generality of functors
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_LINEAR, EXEC_TIMESTEP_BEGIN};
+
+  params.addClassDescription("Examines elements in the geometry to find all positions matching a "
+                             "user-specifed parsed expression criterion.");
+  return params;
+}
+
+ParsedSelectionPositions::ParsedSelectionPositions(const InputParameters & parameters)
+  : Positions(parameters),
+    NonADFunctorInterface(this),
+    BlockRestrictable(this),
+    FunctionParserUtils(parameters),
+    _expression(getParam<std::string>("expression")),
+    _xyzt({"x", "y", "z", "t"}),
+    _functor_names(getParam<std::vector<MooseFunctorName>>("functor_names")),
+    _n_functors(_functor_names.size()),
+    _functor_symbols(getParam<std::vector<std::string>>("functor_symbols"))
+{
+  if (getParam<ExecFlagEnum>("execute_on").contains(EXEC_NONE))
+    paramError("execute_on",
+               "NONE execution flag not supported. Most functors (functions, variables, spatial "
+               "user objects for example) are not initialized at construction.");
+
+  // sanity checks
+  if (!_functor_symbols.empty() && (_functor_symbols.size() != _n_functors))
+    paramError("functor_symbols", "functor_symbols must be the same length as functor_names.");
+
+  // Make sure functors are not x, y, z, or t
+  for (const auto & name : _functor_symbols)
+    if (std::find(_xyzt.begin(), _xyzt.end(), name) != _xyzt.end())
+      paramError("functor_symbols", "x, y, z, and t cannot be used in 'functor_symbols'.");
+  for (const auto & name : _functor_names)
+    if (std::find(_xyzt.begin(), _xyzt.end(), name) != _xyzt.end())
+      paramError(
+          "functor_names",
+          "x, y, z, and t cannot be used in 'functor_names'. Use functor_symbols to disambiguate");
```

Comment:
The second sentence of this error message seems misleading. Seems like you should just delete the second sentence?

Reply:
second sentence is telling the user how to actually do this if needed

i ll try to reword